terraform/aws-https/infra/infra.tf

```diff
@@ -0,0 +1,724 @@
+
+  # enable long-polling
+  receive_wait_time_seconds = 10
+}
+
+output "sqs_queue_name" {
+  value = aws_sqs_queue.file_metadata.name
+}
+
+resource "aws_sqs_queue_policy" "file_metadata_queue_policy" {
+  queue_url = aws_sqs_queue.file_metadata.id
+
+  policy = data.aws_iam_policy_document.file_metadata_queue_policy.json
+}
+
+data "aws_iam_policy_document" "file_metadata_queue_policy" {
+  statement {
+    actions   = ["SQS:SendMessage"]
+    effect    = "Allow"
+    resources = [aws_sqs_queue.file_metadata.arn]
+
+    principals {
+      type        = "Service"
+      identifiers = ["sns.amazonaws.com"]
+    }
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_sns_topic.file_metadata.arn]
+    }
+  }
+}
+
+resource "aws_sns_topic" "file_metadata" {
+  name = "wandb-file-metadata-topic"
+}
+
+resource "aws_sns_topic_policy" "file_metadata_topic_policy" {
+  arn = aws_sns_topic.file_metadata.arn
+
+  policy = data.aws_iam_policy_document.file_metadata_topic_policy.json
+}
+
+data "aws_iam_policy_document" "file_metadata_topic_policy" {
+  statement {
+    sid       = "s3-can-publish"
+    actions   = ["SNS:Publish"]
+    effect    = "Allow"
+    resources = [aws_sns_topic.file_metadata.arn]
+
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+    }
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_s3_bucket.file_storage.arn]
+    }
+  }
+}
+
+resource "aws_sns_topic_subscription" "file_metadata" {
+  topic_arn = aws_sns_topic.file_metadata.arn
+  protocol  = "sqs"
+  endpoint  = aws_sqs_queue.file_metadata.arn
+}
+
+##########################################
+# S3
+##########################################
+
+resource "aws_s3_bucket" "file_storage" {
+  bucket        = "${var.global_environment_name}-wandb-files"
+  acl           = "private"
+  force_destroy = true
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["GET", "HEAD", "PUT"]
+    allowed_origins = ["*"]
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+  }
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+}
+
+output "s3_bucket_name" {
+  value = aws_s3_bucket.file_storage.bucket
+}
+
+output "s3_bucket_region" {
+  value = aws_s3_bucket.file_storage.region
+}
+
+resource "aws_s3_bucket_notification" "file_metadata_sns" {
+  bucket = aws_s3_bucket.file_storage.id
+
+  topic {
+    topic_arn = aws_sns_topic.file_metadata.arn
+    events    = ["s3:ObjectCreated:*"]
+  }
+}
+
+##########################################
+# RDS
+##########################################
+
+resource "aws_db_subnet_group" "metadata_subnets" {
+  name       = "wandb-db-subnets"
+  subnet_ids = aws_subnet.wandb_private[*].id
+}
+
+resource "aws_rds_cluster" "metadata_cluster" {
+  engine               = "aurora-mysql"
```

Comment:
Are you sure the migrations work with Aurora? I got errors with MariaDB and MySQL >= 8. What finally worked fine for me was MySQL 5.7.* though I didn't try Aurora, curious:)

Reply:
Yes it should work :) Sorry for the late response, but this PR is bundled with this tutorial: https://www.youtube.com/watch?v=2ODB4_nnuFE